src/backend/executor/execMain.c

```diff
@@ -2248,6 +2249,8 @@ InitPlan(QueryDesc *queryDesc, int eflags)
 		locallyExecutableSubplans = getLocallyExecutableSubplans(plannedstmt, start_plan_node);
 	}
 
+	initplans = getInitPlans(plannedstmt, start_plan_node);
```

Comment:
Suggest such an optimization:
```suggestion
if (list_length(plannedstmt->subplans)) > 0)
	initplans = getInitPlans(plannedstmt, start_plan_node);
```